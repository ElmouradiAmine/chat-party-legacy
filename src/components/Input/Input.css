.input {
  background-color: var(--color-secondary);
  border-radius: 3px;
  font-size: 1.6rem;
  font-family: 'Poppins', sans-serif;
  border: none;
  padding: 1rem 1.8rem;
  opacity: 0.7;
  color: var(--color-font-primary);
  min-width: 0px;
}

.input:focus {
  outline: none;
  /* border: 1px solid rgba(255,255,255,0.4); */
}
