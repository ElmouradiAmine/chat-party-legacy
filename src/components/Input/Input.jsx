import React from 'react';
import PropTypes from 'prop-types';

import './Input.css';

const Input = ({
  type, className, placeholder, onChange, value, max, maxLength,
}) => (
  <input
    type={type}
    placeholder={placeholder}
    className={`input ${className}`}
    onChange={onChange}
    value={value}
    max={max}
    maxLength={maxLength}
  />
);

Input.propTypes = {
  className: PropTypes.string,
  type: PropTypes.string,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  max: PropTypes.string,
  maxLength: PropTypes.string,

};

Input.defaultProps = {
  className: '',
  type: 'text',
  placeholder: '',
  value: undefined,
  onChange: null,
  max: undefined,
  maxLength: undefined,
};
export default Input;
