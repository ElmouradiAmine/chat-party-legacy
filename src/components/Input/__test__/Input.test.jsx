import React from 'react';
import { render, cleanup } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import Input from '../Input';

beforeEach(cleanup);
describe('Input component', () => {
  const {
    className, type, placeholder, onChange, value, max, maxLength,
  } = {
    className: 'className',
    type: 'text',
    placeholder: 'placeholder',
    value: 'value',
    max: '12',
    maxLength: '12',
    onChange: jest.fn(),
  };

  test('renders correctly', () => {
    render(<Input />);
  });

  test('renders correctly with props', () => {
    const { container } = render(
      <Input
        className={className}
        type={type}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
        max={max}
        maxLength={maxLength}
      />,
    );
    expect(container).not.toBeNull();
    expect(container.firstElementChild?.classList).toContain(className);
    expect(container.firstElementChild.getAttribute('type')).toEqual(type);
    expect(container.firstElementChild.getAttribute('placeholder')).toEqual(
      placeholder,
    );
    expect(container.firstElementChild.value).toEqual(value);
  });

  test('should call onChange', () => {
    const { getByPlaceholderText } = render(
      <Input
        className={className}
        type={type}
        placeholder={placeholder}
        onChange={onChange}
        max={max}
        maxLength={maxLength}
      />,
    );
    const input = getByPlaceholderText(placeholder);
    expect(input).toHaveValue('');
    userEvent.type(input, 'test');

    expect(input).toHaveValue('test');
    expect(onChange).toHaveBeenCalledTimes(4);
  });

  test('matches snapshot', () => {
    const tree = renderer.create(
      <Input
        className={className}
        type={type}
        placeholder={placeholder}
        onChange={onChange}
        value={value}
        max={max}
        maxLength={maxLength}
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
