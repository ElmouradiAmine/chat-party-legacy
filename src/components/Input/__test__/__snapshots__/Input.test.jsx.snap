// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component matches snapshot 1`] = `
<input
  className="input className"
  max="12"
  maxLength="12"
  onChange={[MockFunction]}
  placeholder="placeholder"
  type="text"
  value="value"
/>
`;
